pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -57,11 +57,30 @@ var userRegistriesDir = filepath.FromSlash(".config/containers/registries.conf.d
 // The function errors if the newly created reference is not parsable.
 func (e *Endpoint) rewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
 	refString := ref.String()
+	var newNamedRef string
 	if !refMatchesPrefix(refString, prefix) {
 		return nil, fmt.Errorf("invalid prefix '%v' for reference '%v'", prefix, refString)
 	}
-
-	newNamedRef := strings.Replace(refString, prefix, e.Location, 1)
+	if prefix[0:2] == "*." {
+		if strings.Contains(refString, "/") {
+			firstSlash := strings.Index(refString, "/")
+			repoString := refString[firstSlash:]
+			refString = refString[:firstSlash]
+			for {
+				firstDot := strings.Index(refString, ".")
+				if firstDot == -1 {
+					break
+				}
+				refString = refString[firstDot+1:]
+				if "*."+refString == prefix {
+					newNamedRef = e.Location + repoString
+					break
+				}
+			}
+		}
+	} else {
+		newNamedRef = strings.Replace(refString, prefix, e.Location, 1)
+	}
```

Comment:
This part can probably look much better, but for now it handles the issue:
`A podman pull sub.domain.org/image:latest with subdomainprefix="*.domain.org" and location="registry.com" should be pulling registry.com/image:latest.`